docker-compose.prod.yml

```diff
@@ -32,6 +32,7 @@ services:
      - database.prod.env
     expose:
      -  5432
+    user: postgres
```

Comment:
Won't work if there is no postgres user on your system:

```
➜  match4everything git:(easier-deploy) docker run -u doesneverexist hello-world 
docker: Error response from daemon: unable to find user doesneverexist: no matching entries in passwd file.
ERRO[0000] error waiting for container: context canceled 
```

I tried to get postgres to work well as non-root and I ran into problems all the time, ultimately I abandoned it, but I still think we should support this. Docker-compose.yml supports Env substitution (not from the database.prod.env file unfortunately) see https://docs.docker.com/compose/compose-file/#variable-substitution 

docker-compose up by default uses .env if present for this. We could use var substitution to set a default uid (0) and make it configurable via .env to another uid. We should also switch to using a volume for this, but I think none of those solutions fix the very common error I got:

Run the container once as root by accident and all your permissions will be screwed up. I think this happens with and without volumes. This has happened way too often for me and we should find a way to prevent that

Reply:
Are changing users really a problem for people which are not working on the Docker deployment?

Reply:
Well, I never got  it to run reliably and this PR shouldn't work on a clean system. I think we should make the user configurable and document how to set it up in a way that you can't easily screw with it. The postgres container docs contain some info on this as well under section `Arbitrary --user Notes` on https://hub.docker.com/_/postgres

Reply:
> Won't work if there is no postgres user on your system

That shouldn't be a problem anymore with the `easier-deploy`, right?

Reply:
@maltezacharias

Reply:
Unfortunately still is. In easier-deploy I left out the database container intentionally. I don't know whether the volume solved this because I haven't really investigated how docker handles the file permissions. The main problem is you would need a named postgres user on your docker host which you usually won't have, unless you installed postgres locally

Reply:
Ok, I will just close this PR then

Reply:
To me the key to a solution for this would be that you can
- configure it to use a user you have setup on your system for the job
- make sure that you don't accidentally run the container as root, because that would reset permissions and cause problems later

I think a good solution for this is to create an .env (not to be confused with the env files we load) file where you can specify the host user accounts to use in an env variable and default to root uid if the env variables are not set. Docker auto-loads the .env file and you can then use env var substitution in the compose files. This would also be feasible for the backend. Although I still think gosu would be better there, I read through the security reports and think it would be ok to use, especially because we would still be better off than running as root. The .env approach would probably also work out as well, but running as a user without entries in /etc/passwd can have hard to debug sideeffects in the future.